### ISSUE TITLE
Issue using shared softmax layer.

### ISSUE THREAD

**Issue by alikhodadoost** [monologg/JointBERT#1]
Issue using shared softmax layer, the paper has mentioned using multiplication of slot filling softmax's
https://github.com/monologg/JointBERT/blob/d6e06f2a8910b412e3c3ad84ee533615afdde160/model.py#L52

The original paper code:
https://github.com/MahmoudWahdan/dialog-nlu/blob/7e34ebb3c370abe1044464b9a15c0b445a0a2384/models/joint_bert.py#L65

**Comment by monologg** [comment]
Hello:) Sry for the late reply. 

I've checked the paper again, and I couldn't find that paper has mentioned about `multiplication of softmax`. It seems like you might confused with loss function and objective function. 

I've saw the [line 66](https://github.com/MahmoudWahdan/dialog-nlu/blob/7e34ebb3c370abe1044464b9a15c0b445a0a2384/models/joint_bert.py#L66) of the code, and this one look like `using the hidden state corresponding to the first sub-token as input`, not the multiplication of slot filling softmax.

As I see the github that you mentioned, it also used the same method for loss function ([Cross Entropy](https://github.com/MahmoudWahdan/dialog-nlu/blob/7e34ebb3c370abe1044464b9a15c0b445a0a2384/models/joint_bert.py#L43)) that I used.  The only difference between mine and `dialog-nlu` github is that I use `1 * intent_loss + 1 * slot_loss` and other uses `1 * intent_loss + 3 * slot_loss`. Also the github you mentioned doesn't look like the original author's code.

My assumption might be wrong. If I'm wrong, just let me know:)
